framework/interface/api_handlers.py

```diff
@@ -842,3 +843,65 @@ def delete(self, error_id=None):
             self.get_component("db_error").Delete(error_id)
         except exceptions.InvalidErrorReference:
             raise tornado.web.HTTPError(400)
+
+class ReportExportHandler(custom_handlers.APIRequestHandler):
+    SUPPORTED_METHODS = ['GET']
+
+    def get(self, target_id=None):
+        if not target_id:
+            raise tornado.web.HTTPError(400)
+        try:
+            filter_data = dict(self.request.arguments)  # IMPORTANT!!
+            plugin_outputs = self.get_component("plugin_output").GetAll(filter_data, target_id=target_id, inc_output=True)
+            # Group the plugin outputs to make it easier in template
+            grouped_plugin_outputs = {}
+            for poutput in plugin_outputs:
+                if grouped_plugin_outputs.get(poutput['plugin_code']) is None:
+                    # No problem of overwriting
+                    grouped_plugin_outputs[poutput['plugin_code']] = []
+
+                poutput["rank"] = get_rank(poutput["user_rank"], poutput["owtf_rank"])
+                grouped_plugin_outputs[poutput['plugin_code']].append(poutput)
+            # Needed ordered list for ease in templates
+            grouped_plugin_outputs = collections.OrderedDict(sorted(grouped_plugin_outputs.items()))
+
+            # Get mappings
+            if self.get_argument("mapping", None):
+                mappings = self.get_component("mapping_db").GetMappings(self.get_argument("mapping", None))
+            else:
+                mappings = None
+
+            # Get test groups as well, for names and info links
+            test_groups = {}
+            for test_group in self.get_component("db_plugin").GetAllTestGroups():
+                test_group["mapped_code"] = test_group["code"]
+                test_group["mapped_descrip"] = test_group["descrip"]
+                if mappings:
```

Comment:
Could be shortened:
```python
if mappings and test_group['code'] in mappings:
    test_group["mapped_code"] = mappings[test_group['code']][0]
    test_group["mapped_descrip"] = mappings[test_group['code']][1]
```
Also, it would be more readable to name `[0]` and `[1]`, for instance:
```python
code, description = mappings[test_group['code']]
test_group["mapped_code"] = code
test_group["mapped_descrip"] = description
```